scripts/build_load_data.py

```diff
@@ -0,0 +1,430 @@
+        # https://www.iea.org/data-and-statistics?country=KOSOVO&fuel=Electricity%20and%20heat&indicator=Electricity%20final%20consumption
+        load['KV'] = load['RS'] * (5. / 33.)
+        # Albania (AL) gets the same load curve as Macedonia (MK)
+        # scale parameter selected by energy consumption ratio from IEA Data browser for the year 2017
+        # https://www.iea.org/data-and-statistics?country=ALBANIA&fuel=Electricity%20and%20heat&indicator=Electricity%20final%20consumption
+        load['AL'] = load['MK'] * (6.0 / 7.0)
+        
+        logger.info(f"Missing load data are interpolating linearly and adjusted manual.")
+        logger.info(f"Several load data are missing for 'GR'. Manual processing not possible. Copying the data from another source would be one possibility.")
+
+    return load
+
+
+
+
+
+if __name__ == "__main__":
+
+    if 'snakemake' not in globals():
+        from _helpers import mock_snakemake
+        snakemake = mock_snakemake('build_load_data')
+        rootpath = '..'
+    else:
+        rootpath = '.'
+    
+    configure_logging(snakemake)
+    
+    url = snakemake.config['load']['url']
+    
+    opsd_load = (load_timeseries_opsd(years = slice(*pd.date_range(freq='y', **snakemake.config['snapshots'])[[0,-1]].year.astype(str)),
+                                 fn=url,
+                                 countries = snakemake.config['countries'],
+                                 source = snakemake.config['load']['source']))
+
+    # Convert to naive UTC (has to be explicit since pandas 0.24)
+    opsd_load.index = opsd_load.index.tz_localize(None)
+    
+    # # check the number and lenght of gaps
+    nan_stats = nan_statistics(opsd_load)
+    
+    gap_filling_threshold = snakemake.config['load']['gap_filling_threshold']
+      
+    if nan_stats.consecutive.max() > gap_filling_threshold:        
+        logger.warning(f"Load data contains consecutive gaps of longer than '{gap_filling_threshold}' hours! Check dataset carefully!")
+
+    # adjust gaps and interpolate load data
+    logger.info(f"Gaps of {gap_filling_threshold} hours filled with data from previous week. Smaler gaps interpolated linearly.")
+    opsd_load = opsd_load.apply(fill_large_gaps, gapsize=gap_filling_threshold).interpolate(method='linear', limit=gap_filling_threshold)
+              
+    # adjust gaps manuel
+    if snakemake.config['load']['adjust_gaps_manuel']:
+        logger.info(f"Load data are adjusted manual.")
+        opsd_load = manual_adjustment(load=opsd_load, source=snakemake.config['load']['source'])
+    
+    # check the number and lenght of gaps after adjustment and interpolating
+    nan_stats = nan_statistics(opsd_load)
+    
+    if nan_stats.consecutive.max() > gap_filling_threshold:        
+        logger.warning(f'Load data contains gaps after manuel adjustment. Modify manual_adjustment() function!')
```

Comment:
I think we're almost there. I would restructure this part a bit (sorry for many iterations). 

1. manual corrections if enabled
2. interpolate gaps < gap_size_interpolated (with info) if nan exist
3. fill by weekly shift (with a warning) if nan exist
4. raise a error if nan still exist

Like this we are prioritizing the manual corrections and do the heuristic sanitizing afterwards

Reply:
Should we leave the warning at the beginning that the data set contains gaps? In line 411 & 412
The new order is a good idea!